src/Kestrel.Core/Internal/Http/HttpProtocol.FeatureCollection.cs

```diff
@@ -90,8 +90,21 @@ string IHttpRequestFeature.Scheme
 
         string IHttpRequestFeature.Method
         {
-            get => Method;
-            set => Method = value;
+            get
+            {
+                if (_methodText != null)
+                {
+                    return _methodText;
+                }
+
+                _methodText = HttpUtilities.MethodToString(Method) ?? string.Empty;
+                return _methodText;
+            }
+            set
+            {
+                Method = HttpUtilities.GetKnownMethod(value);
```

Comment:
Out of curiosity, what would happen if we treated every request method set by the app via IHttpRequestFeature.Method as a custom method whether or not it known in order to elide this call to GetKnownMethod?

I don't see how this could be a problem, because none of Kestrel's logic should be predicated on the request method set by the application.

Reply:
So the app couldn't change request behavior based on method? If you changed method from `GET` -> `HEAD` the would mean if the app produced no body, but a content length it would throw with an 500 error.

However that may be better than responding with a `HEAD` response to a `GET` request and then hanging the browser while waiting for data?

Reply:
While the app should be able to change all of the properties of HttpContext to lie to itself, the server must maintain a source of truth so it can properly interact with the client.

Reply:
Changed

Reply:
This change is really messing up the tests :(